lib/ruby_smb/dcerpc/lsarpc.rb

```diff
@@ -1,11 +1,644 @@
+          raise RubySMB::Dcerpc::Error::LsarpcError,
+                "Error returned while querying domain information: "\
+                  "#{WindowsError::NTStatus.find_by_retval(lsar_response.error_status.value).join(',')}"
+        end
+        lsar_response.policy_information
+      end
+
+      def lsar_query_information_policy_2(policy_handle:, information_class:)
+        lsar_request = LsarQueryInformationPolicy2Request.new(
+          policy_handle: policy_handle,
+          information_class: information_class
+        )
+        response = dcerpc_request(lsar_request)
+        begin
+          lsar_response = LsarQueryInformationPolicy2Response.read(response)
+        rescue IOError
+          raise RubySMB::Dcerpc::Error::InvalidPacket, 'Error reading LsarQueryInformationPolicy2Response'
+        end
+        unless lsar_response.error_status == WindowsError::NTStatus::STATUS_SUCCESS
+          raise RubySMB::Dcerpc::Error::LsarpcError,
+                "Error returned while querying domain information: "\
+                  "#{WindowsError::NTStatus.find_by_retval(lsar_response.error_status.value).join(',')}"
+        end
+        lsar_response.policy_information
+      end
+
+      def lsar_close_handle(policy_handle:)
+        lsar_request = LsarCloseHandleRequest.new(
+          policy_handle: policy_handle
+        )
+        response = dcerpc_request(lsar_request)
+        begin
+          lsar_response = LsarCloseHandleResponse.read(response)
+        rescue IOError
+          raise RubySMB::Dcerpc::Error::InvalidPacket, 'Error reading LsarCloseHandleResponse'
+        end
+        unless lsar_response.error_status == WindowsError::NTStatus::STATUS_SUCCESS
+          raise RubySMB::Dcerpc::Error::LsarpcError,
+                "Error returned while closing policy handle: "\
+                  "#{WindowsError::NTStatus.find_by_retval(lsar_response.error_status.value).join(',')}"
+        end
+        lsar_response.policy_handle
+      end
+
+      def lsar_lookup_sids(policy_handle:, sid_enum_buffer:, lookup_level:)
+        lsar_request = LsarLookupSidsRequest.new(
+          policy_handle: policy_handle,
+          sid_enum_buffer: sid_enum_buffer,
+          translated_names: { num_entries: 0, names: nil },
+          lookup_level: lookup_level,
+          mapped_count: 0
+        )
+        response = dcerpc_request(lsar_request)
+        begin
+          lsar_response = LsarLookupSidsResponse.read(response)
+        rescue IOError
+          raise RubySMB::Dcerpc::Error::InvalidPacket, 'Error reading LsarLookupSidsResponse'
+        end
+        unless lsar_response.error_status == WindowsError::NTStatus::STATUS_SUCCESS
+          raise RubySMB::Dcerpc::Error::LsarpcError,
+                "Error returned while looking up SID: "\
+                  "#{WindowsError::NTStatus.find_by_retval(lsar_response.error_status.value).join(',')}"
+        end
+        lsar_response.translated_names
```

Comment:
Instead of returning the `LsaprTranslatedNames` instance, it might be nice to just return an array of strings of the translated names so the caller doesn't need to deal with the structure.